### DISCUSSION TITLE
listing items api or feeds api

### DISCUSSION THREAD

**Discussion by uk-2** [amzn/selling-partner-api-models#3211] [General]
Which does amazon recommend for the following use case 👍 

1.  Creating new listings |  feed api or listing items api ?
2. Updating listings | feed api or listing items api ?
3.  Updating attributes on a listing |  feed api or listing items api ?
4. Adding offers to a listing 

Why have feeds api ?  when you have listing items api ?

Thank you for reading my question

**Comment by michael-land** [comment]
why not use both :-)

**Reply by uk-2** [reply]
Well that's what i was thinking to do , i just wanted to know the implications of speed impact or load imapct on server when serving multiple users

What are you currently doing ?

**Comment by NetRushJacob** [comment]
Depends on the scale you're looking at.

Thousands of individual PATCH operations or a single, bulk feed file to process through.  You'll run through the API quota quite quickly at larger scales, which would lead to slower, more expensive processing.

If you (or your customers) are looking at tens or hundreds of products, I wouldn't put any effort to the product Feeds.
If you are looking at thousands of products that you expect very few changes against, feeds might be handy for the initial push, then API for the changes.

I believe there's also a Json Feed that takes a more bulk approach of the same API models.
